packages/controller-utils/src/util.ts

```diff
@@ -192,18 +218,20 @@ export function hexToText(hex: string) {
   }
 }
 
+export function fromHex(value: string | BN): BN;
+export function fromHex(value: BN4): BN4;
 /**
  * Parses a hex string and converts it into a number that can be operated on in a bignum-safe,
  * base-10 way.
  *
  * @param value - A base-16 number encoded as a string.
  * @returns The number as a BN object in base-16 mode.
  */
-export function fromHex(value: string | BN): BN {
+export function fromHex(value: string | BN | BN4): BN | BN4 {
   if (BN.isBN(value)) {
     return value;
   }
-  return new BN(hexToBN(value).toString(10));
+  return new BN(hexToBN(value as string).toString(10), 10);
```

Comment:
The type-cast here doesn't seem right. TypeScript thinks that `value` is a `string | BN4`.

`BN.isBN` is a type guard that coerces `value` to `BN`. It seems like we need a comparable check for `BN4`:

``` suggestion
  if (BN.isBN(value)) {
    return value;
  }
  if (BN4.isBN(value)) {
    return value;
  }
  return new BN(hexToBN(value).toString(10));
```